cmd/cksum.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * cmd/cksum.c
+ * Purpose: calculate and print POSIX checksums for a specific string.
+ *
+ *  This tool is based on cksum tool from Linux coreutils.
+ *  Is using the same CRC32 lookup table found in crctab.c and same calculations 
+ * as cksum from coreutils to output the same value for the same string input.
+ *  Is necessary for ifupdown-ng when the cksum tools is not avaiable on
+ * a system and we need to generate VRRP names.
+ *
+ * Copyright (c) 2023 EasynetDev (Adrian Ban) <devel@easynet.dev> in the Public Benefit, Inc.
```

Comment:
Your PR claims this is derived from coreutils.  Where is the FSF copyright?  Where is the GPL license?

Please tell me you are not laundering GPL’d code which belongs to the FSF.  But telling me is not enough, at this point I cannot trust this PR until it is proven the code’s legal provenance is clean.

Reply:
> Your PR claims this is derived from coreutils. Where is the FSF copyright? Where is the GPL license?
> 
> Please tell me you are not laundering GPL’d code which belongs to the FSF. But telling me is not enough, at this point I cannot trust this PR until it is proven the code’s legal provenance is clean.

First of all, the previous PR has comments about the license. Now I added ifupdown-ng license to all files. Then you complain about other license? Could you please be more specific what do you really need to be correct from point of view of license?

Reply:
Code which is derived from other sources needs to keep its copyright and license retained.

As a matter of policy, ifupdown-ng requires all code to be ISC licensed (or some equivalent permissive non-copyleft license).  This means that ifupdown-ng cannot accept code derived from GNU coreutils as it is under GPL and thus copyleft.

But simply changing the copyright and license of the code to fix this problem is *illegal*.  It is not the correct way to solve this problem, and we can't accept that either.

There are several public domain crc32/crc32c routines which you could use, which could be brought into ifupdown-ng.

Reply:
The formula for the CRC32 is the same. I look to several CRC32 implementations and the base of the calculation is almost the same.
I took this tool as the foundation for ifcrc32 to have the same output as crc32 as coreutils.

I rewrite plenty of the code. Is not reading anymore a whole file, just a buffer. Then the calculation I had to rewrite to use the buffer instead chunks as crc32 is using.

The hashes are the same as crc32, but that file can be generated by crc32 if you compile with a specific - D parameter and you run it and you get the array.

What I don't understand is were I'm laundering the license.

Do you need specific parts I used to rewrite the code to adapt it for this specific purpose?

Reply:
Hi @kaniini 

The polynomial formula is not coreutils implementation. Is just from here:
https://en.wikipedia.org/wiki/Cyclic_redundancy_check

The numbers generated in cksum.h you can find, a part of them, also in that link. Example: 0x04C11DB7

All crc32/crc32c routines has the similar polynomial. The difference is done by how the last 0-7 bytes are calculated in the CRC32.